catroidTest/src/org/catrobat/catroid/uitest/content/brick/PlaceAtBrickTest.java

```diff
@@ -94,6 +97,46 @@ public void testPlaceAtBrick() throws InterruptedException {
 				R.id.brick_place_at_edit_text_y, yPosition, Brick.BrickField.Y_POSITION, placeAtBrick);
 	}
 
+	public void testBehaviorOfUndoAndRedoButton(){
+		solo.clickOnView(solo.getView(R.id.brick_place_at_edit_text_x));
+		solo.clickOnView(solo.getView(R.id.formula_editor_keyboard_5));
+		View undo = solo.getView(R.id.menu_undo);
+		solo.waitForCondition(new Condition() {
+			@Override
+			public boolean isSatisfied() {
+				return solo.getView(R.id.menu_undo).isEnabled();
+			}
+		}, 500000);
```

Comment:
The lambda representation of this code, which Android Studio automatically performs, looks really nice. I like that. :+1:
However, 500000 is a reeeally large timeout.